src/hugr.rs

```diff
@@ -340,6 +384,12 @@ pub enum HugrError {
     /// An error occurred while manipulating the hierarchy.
     #[error("An error occurred while manipulating the hierarchy.")]
     HierarchyError(#[from] portgraph::hierarchy::AttachError),
+    /// The node doesn't exist.
```

Comment:
This doesn't seem to be used? (Also I wonder about the utility of naming the single field within, but there.)

Reply:
I prefer to use named fields when the order is not relevant (what does `x.0` mean?). But I have no strong opinion on this.

Reply:
I changed it into a tuple struct for consistency with the other errors.